### ISSUE TITLE
Ty misses type inference for literals assigned to `Sequence`

### ISSUE THREAD

**Issue by breid48** [astral-sh/ty#3489]
### Summary

The following code snippet passes on all of Mypy, Pyright and Pyrefly, but fails with Ty.

We depend on this inference succeeding because we represent certain internal operations in our system using
similar constructs of literals inside tuples. It would be great if we could have parity with the other type checkers in this regard!

```python
from typing import Literal, Sequence


A = tuple[Literal[1], int]
B = tuple[Literal[2], int]
ABList = list[A | B]
ABSequence = Sequence[A | B]

ab: ABList = [(1, 5)]  # OK
ab_seq: ABSequence = [(1, 5)]  # error: Object of type `list[tuple[int, int]]` is not assignable to `Sequence[tuple[Literal[1], int] | tuple[Literal[2], int]]`: Incompatible value of type `list[tuple[int, int]]`

```

Playground: https://play.ty.dev/4e324983-ef76-44e4-a60b-47113f8c6f29

### Version

0.0.37

**Comment by carljm** [comment]
Yeah, this looks like something we should support. Thanks for reporting!

**Comment by charliermarsh** [comment]
I _think_ this is fixed by https://github.com/astral-sh/ruff/pull/23186.

**Comment by ibraheemdev** [comment]
The asssignment in the repro should succeed regardless of our literal promotion heuristics. It seems that somewhere down the line the type context isn't being propagated.

Aside: I'm not exactly sure if this will be indirectly fixed by https://github.com/astral-sh/ruff/pull/23186, because the conclusion to https://github.com/astral-sh/ty/issues/136 was different to what was discussed on that PR. I *think* it still makes sense to avoid promotion here because of the explicit annotation, which means you would end up with a `list[Literal[..]]` as the RHS of an assignment annotated as `Sequence[..]`, which is not a very useful `list`, but probably a more useful type to the user than the promoted list given that they explicitly annotated it as a `Sequence`.

**Comment by carljm** [comment]
It looks like this was fixed by https://github.com/astral-sh/ruff/pull/27311
